webapp/public/javascripts/controllers.js

```diff
@@ -100,20 +117,27 @@ lunchrControllers.controller('HomeMatchedController', ['$scope', 'authService',
 lunchrControllers.controller('HomePageController', ['$scope', '$http', '$state', 'authService',
     function ($scope, $http, $state, authService) {
         var currentUser = authService.currentUser();
-        if(currentUser.matchedWith){
-            $state.go('home.matched');
-        }
-        if(currentUser.wantsToBeMatched){
-            $state.go('home.matching');
-        }
-
-        $scope.name = (currentUser.firstname + " " + currentUser.lastname);
-        $scope.match = function () {
-            $state.go('home.matching');
+        $scope.init = function () {
+            return checkUser(currentUser);
         };
 
-        $scope.editInfo = function () {
-            ;
-        };
+        if(currentUser) {
```

Comment:
I was getting html errors when I wasn't logged in because when currentUser is null the JS can't parse currentUser.matchedwith and other variables in the user object.
chechUser is the function that does the actual checking, init is declared in every controller that uses login checking, and init returns true or false so the jade can tell whether to display the page or the error